RUFAS/routines/field/field/field.py

```diff
@@ -644,22 +578,206 @@ def _execute_manure_application(
+            dry_matter_mass=manure_supplied.dry_matter,
+            dry_matter_fraction=manure_supplied.dry_matter_fraction,
+            field_coverage=field_coverage,
+            nitrogen=supplied_nitrogen,
+            phosphorus=supplied_phosphorus,
+            potassium=None,
+            application_depth=application_depth,
+            surface_remainder_fraction=surface_remainder_fraction,
+            year=year,
+            day=day,
+            output_name="manure_application",
+        )
 
-        if unmet_nitrogen_demand == 0.0 and unmet_phosphorus_demand == 0.0:
+        return supplied_nitrogen, supplied_phosphorus, application_depth, surface_remainder_fraction
+
+    def _validate_application_depth_and_fraction(
+        self,
+        application_depth: float,
+        surface_remainder_fraction: float,
+        year: int,
+        day: int,
+    ) -> tuple[float, float]:
+        """
+        Validates the application depth and surface remainder fraction for a manure application.
+
+        Parameters
+        ----------
+        application_depth : float
+            Depth at which fertilizer is injected into the soil (mm).
+        surface_remainder_fraction : float
+            Fraction of fertilizer applied that remains on the soil surface after application (unitless).
+        year : int
+            Calendar year in which this manure application occurs.
+        day : int
+            Julian day on which this manure application occurs.
+
+        Returns
+        -------
+        tuple[float, float]
+            The validated application depth and surface remainder fraction.
+
+        Raises
+        ------
+        ValueError
+            If the soil layers are not initialized or if the bottom depth is not set for the last soil layer.
+        """
+        error_name = "manure_application_error"
+
+        if (application_depth == 0.0 and surface_remainder_fraction != 1.0) or (
+            application_depth > 0.0 and surface_remainder_fraction == 1.0
+        ):
+            self._record_nutrient_application_error(
+                application_depth, surface_remainder_fraction, error_name, year, day
+            )
+            return 0.0, 1.0
+
+        soil_layers = self.soil.data.soil_layers
+        if not soil_layers:
+            raise ValueError("soil_layers is not initialized")
+
+        bottom_layer = soil_layers[-1]
+        if bottom_layer.bottom_depth is None:
+            raise ValueError("bottom_depth is not set for the last soil layer")
+
+        max_depth = bottom_layer.bottom_depth
+        if application_depth > max_depth:
+            self._record_nutrient_application_error(application_depth, None, error_name, year, day)
+            application_depth = max_depth
+
+        return application_depth, surface_remainder_fraction
+
+    def _handle_unmet_nutrients(
+        self,
+        requested_n: float,
+        requested_p: float,
+        supplied_n: float,
+        supplied_p: float,
```

Comment:
```suggestion
        requested_nitrogen: float,
        requested_phosphorus: float,
        supplied_nitrogen: float,
        supplied_phosphorus: float,
```
Consider renaming these variables to ensure consistency with other parts of the code.